### PR TITLE
[doc][monitoring][txn] add doc for transaction metrics

### PR DESCRIPTION
**Pull request by tjiuming**
### Motivation

add doc for transaction metrics(https://github.com/apache/pulsar/pull/15140).

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [ ] `no-need-doc` 
(Please explain why)
  
- [x] `doc` 
(Your PR contains doc changes)

- [ ] `doc-added`
(Docs have been already added)

### CODE REVIEW

**Comment by tjiuming** [comment]
image
image

**Comment by Anonymitaet** [comment]
@momo-jun can you help review this PR? Thanks!

File: site2/docs/reference-metrics.md [hunk 1]
```diff
@@ -638,3 +641,10 @@ All the transaction metrics are labelled with the following labels:
...
+| pulsar_txn_tb_client_abort_failed | Counter | Number of `transaction buffer client`'s failed operations to abort transaction. |
```
Comment (momo-jun) [inline site2/docs/reference-metrics.md:644]:
```suggestion
| pulsar_txn_tb_client_abort_failed | Counter | The number of `transaction buffer client`'s failed operations to abort transactions. |
```

This change applies to all others.

File: site2/docs/reference-metrics.md [hunk 2]
```diff
@@ -638,3 +641,10 @@ All the transaction metrics are labelled with the following labels:
...
+| pulsar_txn_tb_client_commit_failed | Counter | Number of `transaction buffer client`'s failed operations to commit transaction. |
+| pulsar_txn_tb_client_abort_latency | Summary | Latency of `transaction buffer client`'s operations to abort transaction. |
+| pulsar_txn_tb_client_commit_latency | Summary | Latency of `transaction buffer client`'s operations to commit transaction. |
+| pulsar_txn_tb_client_pending_requests | Counter | Number of `transaction buffer client`'s pending requests. |
+| pulsar_txn_tp_committed_count | Counter | Number of consumer's committed transactions. |
```
Comment (momo-jun) [inline site2/docs/reference-metrics.md:649]:
```suggestion
| pulsar_txn_tp_committed_count | Counter | The number of committed transactions for consumers. |
```

File: site2/docs/reference-metrics.md [hunk 3]
```diff
@@ -638,3 +641,10 @@ All the transaction metrics are labelled with the following labels:
...
+| pulsar_txn_tb_client_abort_failed | Counter | Number of `transaction buffer client`'s failed operations to abort transaction. |
+| pulsar_txn_tb_client_commit_failed | Counter | Number of `transaction buffer client`'s failed operations to commit transaction. |
+| pulsar_txn_tb_client_abort_latency | Summary | Latency of `transaction buffer client`'s operations to abort transaction. |
+| pulsar_txn_tb_client_commit_latency | Summary | Latency of `transaction buffer client`'s operations to commit transaction. |
+| pulsar_txn_tb_client_pending_requests | Counter | Number of `transaction buffer client`'s pending requests. |
```
Comment (momo-jun) [inline site2/docs/reference-metrics.md:648]:
```suggestion
| pulsar_txn_tb_client_pending_requests | Counter | The number of pending requests for `transaction buffer client`. |
```

File: site2/docs/reference-metrics.md [hunk 4]
```diff
@@ -638,3 +641,10 @@ All the transaction metrics are labelled with the following labels:
...
+| pulsar_txn_tb_client_abort_failed | Counter | Number of `transaction buffer client`'s failed operations to abort transaction. |
+| pulsar_txn_tb_client_commit_failed | Counter | Number of `transaction buffer client`'s failed operations to commit transaction. |
+| pulsar_txn_tb_client_abort_latency | Summary | Latency of `transaction buffer client`'s operations to abort transaction. |
```
Comment (momo-jun) [inline site2/docs/reference-metrics.md:646]:
```suggestion
| pulsar_txn_tb_client_abort_latency | Summary | The latency of aborting transactions for `transaction buffer client`. |
```

File: site2/docs/reference-metrics.md [hunk 5]
```diff
@@ -217,6 +217,9 @@ All the topic metrics are labelled with the following labels:
 | pulsar_compaction_latency_le_* | Histogram | The compaction latency with given quantile. <br> Available thresholds: <br><ul><li>pulsar_compaction_latency_le_0_5: <= 0.5ms </li><li>pulsar_compaction_latency_le_1: <= 1ms</li><li>pulsar_compaction_latency_le_5: <= 5ms</li><li>pulsar_compaction_latency_le_10: <= 10ms</li><li>pulsar_compaction_latency_le_20: <= 20ms</li><li>pulsar_compaction_latency_le_50: <= 50ms</li><li>pulsar_compaction_latency_le_100: <= 100ms</li><li>pulsar_compaction_latency_le_200: <= 200ms</li><li>pulsar_compaction_latency_le_1000: <= 1s</li><li>pulsar_compaction_latency_le_overflow: > 1s</li></ul> |
 | pulsar_compaction_compacted_entries_count | Gauge | The total number of the compacted entries. |
 | pulsar_compaction_compacted_entries_size |Gauge  | The total size of the compacted entries. |
+| pulsar_txn_tb_active_count | Gauge | Number of active transactions of the topic. |
```
Comment (momo-jun) [inline site2/docs/reference-metrics.md:220]:
```suggestion
| pulsar_txn_tb_active_count | Gauge | The number of active transactions on this topic. |
```
This change applies to others.
Just wonder this metric is "Gauge" while the other two are "Count"? They are all about counting.

Comment (tjiuming) [reply site2/docs/reference-metrics.md]: It cannot be a counter. counter means always increasing, but currently active txns could increase or decrease

File: site2/docs/reference-metrics.md [hunk 6]
```diff
@@ -218,6 +218,9 @@ All the topic metrics are labelled with the following labels:
 | pulsar_compaction_latency_le_* | Histogram | The compaction latency with given quantile. <br /> Available thresholds: <br /><ul><li>pulsar_compaction_latency_le_0_5: <= 0.5ms </li><li>pulsar_compaction_latency_le_1: <= 1ms</li><li>pulsar_compaction_latency_le_5: <= 5ms</li><li>pulsar_compaction_latency_le_10: <= 10ms</li><li>pulsar_compaction_latency_le_20: <= 20ms</li><li>pulsar_compaction_latency_le_50: <= 50ms</li><li>pulsar_compaction_latency_le_100: <= 100ms</li><li>pulsar_compaction_latency_le_200: <= 200ms</li><li>pulsar_compaction_latency_le_1000: <= 1s</li><li>pulsar_compaction_latency_le_overflow: > 1s</li></ul> |
 | pulsar_compaction_compacted_entries_count | Gauge | The total number of the compacted entries. |
 | pulsar_compaction_compacted_entries_size |Gauge  | The total size of the compacted entries. |
+| pulsar_txn_tb_active_count | Gauge | The number of active transactions on this topic. |
+| pulsar_txn_tb_aborted_count | Counter | The number of aborted transactions on the topic. |
+| pulsar_txn_tb_committed_count | Counter | The number of committed transaction on the topic. |
```
Comment (momo-jun) [inline site2/docs/reference-metrics.md:223]:
```suggestion
| pulsar_txn_tb_committed_count | Counter | The number of committed transactions on the topic. |
```

File: site2/docs/reference-metrics.md [hunk 7]
```diff
@@ -660,3 +663,10 @@ All the transaction metrics are labelled with the following labels:
...
+| pulsar_txn_tb_client_abort_failed | Counter | The number of failures to abort transactions for `transaction buffer client`. |
+| pulsar_txn_tb_client_commit_failed | Counter | The number of failures to commit transaction for `transaction buffer client`. |
+| pulsar_txn_tb_client_abort_latency | Summary | The latency of aborting transactions for `transaction buffer client`. |
+| pulsar_txn_tb_client_commit_latency | Summary | The latency of committing transactions for `transaction buffer client`. |
+| pulsar_txn_tb_client_pending_requests | Counter | The number of pending requests for `transaction buffer client`. |
+| pulsar_txn_tp_committed_count | Counter | The number of committed transactions for consumers. |
```
Comment (codelipenghui) [inline site2/docs/reference-metrics.md:671]:
```suggestion
| pulsar_txn_tp_committed_count | Counter | The number of committed transactions for pending ack store. |
```

File: site2/docs/reference-metrics.md [hunk 8]
```diff
@@ -660,3 +663,10 @@ All the transaction metrics are labelled with the following labels:
...
+| pulsar_txn_tp_aborted_count | Counter | The number of aborted transactions for consumers. |
```
Comment (codelipenghui) [inline site2/docs/reference-metrics.md:672]:
```suggestion
| pulsar_txn_tp_aborted_count | Counter | The number of aborted transactions for pending ack store. |
```

**Comment by tjiuming** [comment]
https://github.com/apache/pulsar/pull/15140 merged, please help review/approve/merge the PR.
@Anonymitaet @momo-jun @codelipenghui

**Review by Anonymitaet — APPROVED** [review]

**Review by momo-jun — APPROVED** [review]
LGTM.

**Comment by tjiuming** [comment]
@codelipenghui could this PR be merged?

**Pull request merged**